src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -943,6 +944,95 @@ private Task ConfigFileDeserializationValidationHelper(string jsonString)
             return Verify(runtimeConfig);
         }
 
+        /// <summary>
+        /// Helper method to parse the config json string and set the connection string field to a given string.
+        /// </summary>
+        /// <param name="configJsonString">Config Json string</param>
+        /// <param name="newConnectionString">Value to update the connection string with</param>
+        /// <returns>Config Json string with connection string set to the given value</returns>
+        private static string UpdateConnectionStringWithCustomValue(string configJsonString, string newConnectionString)
+        {
+            JObject configJson = JObject.Parse(configJsonString);
+            configJson["data-source"]["connection-string"] = newConnectionString;
+            return configJson.ToString();
+        }
+
+        /// <summary>
+        /// Validates that the checked-in reference config file for MsSql (dab-config.MsSql.reference.json) reflects the latest format with which
+        /// the CLI generates the config files.
+        /// The pipeline that runs the integration tests for MsSql, generates a config file (dab-config.MsSql.json) and uses this for all the tests.
+        /// This test compares these two files to make sure that the reference config file is up to date.
+        /// </summary>
+        [TestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        public void TestReferenceConfigFileIsInLatestFormatForMsSql()
+        {
+            string referenceConfigJson = File.ReadAllText($"{CONFIGFILE_NAME}.{MSSQL_ENVIRONMENT}.{REFERENCE_CONFIG_FILE_SUFFIX}{CONFIG_EXTENSION}");
```

Comment:
Is the mssql config file only present in the file system when the `[TestCategory(TestCategory.MSSQL)]` attribute is added to this test method?

Reply:
Instead of different functions for all the dbs, can we use DataRow for each db type and run these tests at the end of Build similar to the Unit Tests?